### DISCUSSION TITLE
Is VSCode use the tree-sitter to parse language?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [tree-sitter/tree-sitter#3519] [Q&A] [answered]
I want to use tree-sitter to implement the function: Extract classes, properties, and methods from the code file. 
[![enter image description here][1]][1]

so I check out the source code of VSCode:

 I see: [treeSitterParserService.ts](https://github.com/microsoft/vscode/blob/2d16be99691d24769f02d417ecc32c784a63edc4/src/vs/editor/browser/services/treeSitter/treeSitterParserService.ts)

1、
I found it only used:
`@vscode/tree-sitter-wasm`  with no grammar.

and I also found the `@vscode/tree-sitter-wasm` is not `Tree-sitter`, it is used to [build WebAssembly files](https://github.com/microsoft/vscode-tree-sitter-wasm/tree/main).

> This repository contains scripts and build pipelines for building the Tree-Sitter and Tree-Sitter grammar WebAssembly files used by VS Code. 

2、How VSCode can parse all programming languages? 
Because there is no other language's tree-sitter-grammer was installed in VSCode, then it can parse all languages?

in the [official document](https://github.com/tree-sitter/node-tree-sitter) it state:
> you'll need a Tree-sitter grammar for the language you want to parse.

  [1]: https://i.sstatic.net/M62muC4p.png

**Comment by amaanq** [comment, marked as answer]
No, VSCode uses TextMate.

**Reply by sogaiu** [reply]
I think this used to be (and still may be) true, but there seems to be [some activity indicating that things might change](https://github.com/microsoft/vscode/issues/207416).

**Reply by amaanq** [reply]
Oh that's really interesting - thanks for the link :)
